src/components/locations/components/LocationById.js

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+    Icon, Card, Carousel, Row, Col, Button, Typography, Tag
+} from 'antd';
+import { perWeek, naira, LOCAL_GOVERNMENT, ADDRESS, STATE } from '../constants';
+import { getLocationsByID } from '../actions';
+
+const LocationById = ({ match, history }) => {
+    const dispatch = useDispatch();
+    const { locationById, locationLoading } = useSelector(state => state.location);
+    const { name, price, lga, trafficRate, address, state, images } = locationById;
+    const { params } = match;
+    const { id: locationId } = params;
+    useEffect(() => {
+        dispatch(getLocationsByID(locationId));
+    }, [dispatch, locationId]);
+    const renderAmount = text => (
+        <Typography.Text type="secondary">
+            <del>{naira}</del>
+            <strong>
+                {text}
+            </strong>
+            <span>
+                {perWeek}
+            </span>
+        </Typography.Text>
+    );
+    const renderImages = locImages => locImages.map((element, i) => (
+        !locationLoading ? (
+            <img
+                key={element}
+                alt={i}
+                src={element}
+                style={{
+                    objectFit: 'cover',
+                }}
+            />
+        ) : ''
+
+    ));
+    const renderTag = text => {
+        let color;
+        let rateText;
+        if (parseInt(text, 10) <= 200) {
+            color = 'volcano';
+            rateText = 'LOW';
+        } else if (parseInt(text, 10) <= 500) {
+            color = 'orange';
+            rateText = 'MEDIUM';
+        } else {
+            color = 'green';
+            rateText = 'HIGH';
+        }
+        return (
+            <Tag color={color}>
+                {rateText}
+            </Tag>
+        );
+    };
```

Comment:
this function is duplicated code, Turn it into a helper function that returns `{ color, rateText }`